shtdlib.sh

```diff
@@ -661,6 +661,54 @@ run_dir="${run_dir:-$(dirname "${script_full_path}")}"
 # Default is to clean up after ourselves
 cleanup="${cleanup:-true}"
 
+# Create NSS Wrapper passwd and group files
+# Accepts 3 optional arguments, username, group and home directory
+# Defaults to bob, builders and a temporary directory
+# Note that if a home directory is specified and it's temporary it will need to
+# be removed/cleaned up by the code calling this function
+function init_nss_wrapper {
+    umask_decorator_mask=${NSS_WRAPPED_FILE_MASK:-0002}
+    umask_decorator "${FUNCNAME[0]}" "${@:-}" && return
+
+    debug 8 'Initializing NSS Wrapper'
+    assert test -n "${BUILD_GUID}"
+
+    export TMP_USER="${1:-bob}"
+    export TMP_GROUP="${2:-builders}"
+    tmp_passwd_file="$(mktemp)" && add_on_exit "rm -f '${tmp_passwd_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_passwd_file}"
```

Comment:
Can we add a suffixes to these paths to make troubleshooting easier? 

e.g. 

```
tmp_passwd_file="$(mktemp)-passwd" && ...
```

Reply:
That's a good idea, I've switched to using templates and including the filename and pid.

Reply:
Whats the `XXXXXXXXXX` for? Just so we know its our stuff?